scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -562,6 +562,18 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
     }
   }
 
+  // corresponding quasiquote (rank 1) for expressions like `{..$smth}`
+  def constructR1Quasi[T <: Tree : AstMetadata](ctor: => Tree): List[T] = {
```

Comment:
That was my first approximation of this method. But I didn't get any fruitful results, because additional boolean parameter brings too much noise, and `ellipsis` becomes almost unreadable.

For example, consider `val q"for (..$enumerators) yield $expr" = q"for (a <- as; b <- bs) yield foo(x, y)"`.
Before `constructR1Quasi`/`ellipsis` current token is `$enumerators`, and that is not applicable to `ellipsis` and would require some step back in iterator. Or issue with `T/Tree` below.

So, `constructR1Quasi` can be viewed as more suitable method for several situations.